### PR TITLE
Add getipaddrs

### PR DESCRIPTION
**Pull request by samuel-massinon**
`getipaddr` would only return the first IP address it would find.
`getipaddrs` will return all the IP addresses it finds.

We ran into an issue when running on a machine with multiple IP addresses. The IP returned by `getipaddr` would not be the one we needed. This function solved the issue for us.

### CODE REVIEW

File: stdlib/Sockets/src/addrinfo.jl
Comment (ararslan) [inline stdlib/Sockets/src/addrinfo.jl]: The convention for this in docstrings is to list the type itself rather than an instance of it, so this should be
```suggestion
    getipaddrs() -> Vector{IPv4}
```

**Review by ararslan — COMMENTED** [review]
Can you also add this to the Sockets section of the documentation alongside `getipaddr`?

**Review by ararslan — APPROVED** [review]
Looks good to me, but it would be good to have another set of eyes on this from someone who knows more about these things.

EDIT: Needs an entry in NEWS.md.

**Review by omus — APPROVED** [review]
Looks good to me. I wouldn't mind if `getipaddr()` used `getipaddrs()` but to do that efficiently we'd probably want to make `getipaddrs` return an generator.

File: stdlib/Sockets/src/addrinfo.jl
Comment (fredrikekre) [inline stdlib/Sockets/src/addrinfo.jl]:
```suggestion

!!! compat "Julia 1.2"
    This function is available as of Julia 1.2.
    
```

File: stdlib/Sockets/src/addrinfo.jl
Comment (vtjnash) [inline stdlib/Sockets/src/addrinfo.jl]: dropping these seems odd to me. seems like these should probably be returned (with each entry being a namedtuple), or a perhaps configurable by function argument

**Comment by vtjnash** [comment]
I think I'd like this to return a vector of namedtuples with all of the various properties exposed (https://nodejs.org/api/os.html#os_os_networkinterfaces)

**Comment by samuel-massinon** [comment]
@vtjnash 

Would this be better in a different function instead of this one? I like the idea of `getipaddrs` just returning a list of values `getipaddr` should return.

If we want to go down that path, maybe a function named `get_network_interfaces` could return somthing like you posted.

**Comment by StefanKarpinski** [comment]
> If we want to go down that path, maybe a function named `get_network_interfaces` could return somthing like you posted.

Please yes, it would be really unexpected if `getipaddr` returns one IP address and `getipaddrs` doesn't return a list of the same. If we want a `Sockets.getifaces` function that returns a bunch of structs or named tuples, that would be fine, but that's a different feature entirely.

File: stdlib/Sockets/src/addrinfo.jl
Comment (StefanKarpinski) [reply stdlib/Sockets/src/addrinfo.jl]: The current behavior seems like it is a reasonable default, so we can add an option to list internal interfaces as well as external ones at a future time, although it would not be a bad addition to make in this PR given that this new feature won't land in a release until 1.2 in the spring.

**Comment by samuel-massinon** [comment]
Can I get this merged/re-reviewed?

**Comment by StefanKarpinski** [comment]
There's still ongoing discussion about whether to include internal IPs by default. Once that's decided, we can go ahead and merge this. @vtjnash, what are your thoughts on that issue?

**Comment by vtjnash** [comment]
OK, I'm convinced. I just had a naming question: should we go with `getallipaddr`, to parallel with `getalladdrinfo`?

**Comment by StefanKarpinski** [comment]
No, I think `getipaddrs` is better—I kind of wish that `getalladdrinfo` had been called `getaddrinfo`.

File: stdlib/Sockets/src/addrinfo.jl
Comment (samuel-massinon) [reply stdlib/Sockets/src/addrinfo.jl]: Does my latest changes resolve this issue?

**Comment by samuel-massinon** [comment]
Is this good to merge? Have all the issues been resolved?

**Comment by omus** [comment]
I think this is good to go. I'll merge this at the end of my day (6 hours from now) unless there are objections.

**Comment by ararslan** [comment]
Thanks, Sam! Nice work 👍

**Comment by staticfloat** [comment]
This seems to have broken the buildbots, likely because of some weirdness within the buildbot setup.  In particular, the failing test is:

```
Error in testset Sockets:
Test Failed at /buildworker/worker/tester_linux32/build/share/julia/stdlib/v1.2/Sockets/test/runtests.jl:427
  Expression: getipaddrs(true) >= getipaddrs()
   Evaluated: Sockets.IPv4[ip"127.0.0.1", ip"172.19.0.4", ip"172.18.0.4"] >= Sockets.IPv4[ip"172.19.0.4", ip"172.18.0.4"]
```

It looks to me like there might be a problem when a machine has multiple IP addresses?

**Comment by samuel-massinon** [comment]
The problem is how I compared the lists. 
I was thinking that `getipaddrs(true) >= getipaddrs()` would test if `getipaddrs()` is a subset of `getipaddrs(true)`. But I guess the order of the list matters.

I think we would want to replace the test with
```julia
issubset(getipaddrs(), getipaddrs(true))
```

**Pull request merged**
